### PR TITLE
Monika/coin dropdown

### LINKED ISSUE #276: Multiple denom support not working as expected
@mdyring from [e-Money](https://github.com/e-money) has deployed with the latest version of Big Dipper but the multiple denom is still not being displayed correctly.

IMAGE 2020-02-07 17:57:16

Here is the settings he is currently using
```
{
    "public":{
        "chainName": "e-Money Testnet 5",
        "chainId": "${CHAIN_ID}",
        "gtm": "",
        "slashingWindow": 10000,
        "uptimeWindow": 250,
        "initialPageSize": 30,
        "bech32PrefixAccAddr": "emoney",
        "bech32PrefixAccPub": "emoneypub",
        "bech32PrefixValAddr": "emoneyvaloper",
        "bech32PrefixValPub": "emoneyvaloperpub",
        "bech32PrefixConsAddr": "emoneyvalcons",
        "bech32PrefixConsPub": "emoneyvalconspub",
        "stakingDenom": "ungm",
        "mintingDenom": "ungm",
        "stakingFraction": 1000000,
        "gasPrice": 0.02,
        "coingeckoId": "false",
        "coins": [
            {
                "stakingDenom": "NGM",
                "stakingDenomPlural": "NGMs",
                "mintingDenom": "ungm",
                "staking": true,
                "stakingFraction": 1000000
            },
            {
                "stakingDenom": "eCHF",
                "stakingDenomPlural": "eCHF",
                "mintingDenom": "echf",
                "staking": false,
                "stakingFraction": 1000000
            },
            {
                "stakingDenom": "eEUR",
                "stakingDenomPlural": "eEUR",
                "mintingDenom": "eeur",
                "staking": false,
                "stakingFraction": 1000000
            },
            {
                "stakingDenom": "eJPY",
                "stakingDenomPlural": "eJPY",
                "mintingDenom": "ejpy",
                "staking": false,
                "stakingFraction": 1000000
            },
            {
                "stakingDenom": "eUSD",
                "stakingDenomPlural": "eUSD",
                "mintingDenom": "eusd",
                "staking": false,
                "stakingFraction": 1000000
            } 
        ]
    },
    "genesisFile": "${GENESIS_URI}",
    "remote":{
        "rpc":"https://${NETWORK}.validator.network:443",
        "lcd":"https://${NETWORK}.validator.network:443/light"
    },
    "debug": {
        "startTimer": true,
        "readGenesis": true
    },
    "params":{
        "startHeight": 0,
        "defaultBlockTime": 60000,
        "blockInterval": 1000,
        "consensusInterval": 1000,
        "statusInterval":7500,
        "signingInfoInterval": 1800000,
        "proposalInterval": 86400000,
        "missedBlocksInterval": 60000,
        "delegationInterval": 900000
    }
}
```

We have a node running and the RPC of the node is at `tcp://34.87.22.68:26657`.

### PR DESCRIPTION
**Pull request by MonikaCat**
## Description

Fixes #276

## Checklist
- [x] Targeted PR against the correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Added an entry to the `CHANGELOG.md` file.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,13 +1,7 @@
...
 
 ## [Unreleased]
-* Changed the structure of coins.js and settings.json 
-* Update Meteor to v1.9
-* [#267] Fixed Fee Decimal Places
-* [#263] Fixed Tx fee to be displayed in Minting or Staking Denom depending on the fee amount
-* [#262] Fixed Validator Popover in Account Section 
-* [#260] Fixed Token unit to display the correct Denom Name/Type  
-* [#227] Added additional info to be displayed for Parameter Change Proposal and Community Pool Spend Proposal
+* [#276] Added Coin Selection Dropdown in Account Section to support multiple denomination
```
Comment (kwunyeung) [inline CHANGELOG.md]: The changelog should be kept.

File: imports/ui/accounts/Account.jsx [hunk 2]
```diff
@@ -61,15 +65,18 @@ export default class AccountDetails extends Component{
...
-                // console.log(result);
+                let totalAmount = [];
+                let amountAvailable = [];
+
                 if (result.available){
-                    const amount = result.available.amount || '0';
-                    const denomType  = result.available.denom;
 
+                    amountAvailable = cloneDeep(result.available);
+                    totalAmount = cloneDeep(result.available);
```
Comment (kwunyeung) [inline imports/ui/accounts/Account.jsx]: What is the `totalAmount` is for?

File: imports/ui/accounts/Account.jsx [hunk 3]
```diff
@@ -80,54 +87,106 @@ export default class AccountDetails extends Component{
...
                             delegated: this.state.delegated+parseFloat(amount),
-                            total: this.state.total+parseFloat(amount)
                         })
                     }, this)
+
+                    totalAmount.forEach((total, i) => {
+                    if(total.denom === Meteor.settings.public.bondDenom )
+                    {
+                     let addValue = parseFloat(totalAmount[i].amount) + parseFloat(this.state.delegated);
+                     totalAmount[i].amount = addValue
```
Comment (kwunyeung) [inline imports/ui/accounts/Account.jsx]: This can be done without declaring the `addValue` variable.

File: imports/ui/accounts/Account.jsx [hunk 4]
```diff
@@ -80,54 +87,106 @@ export default class AccountDetails extends Component{
...
+                    }
+                    }, this)
+
+                    this.setState({
+                             total: [...totalAmount]
```
Comment (kwunyeung) [inline imports/ui/accounts/Account.jsx]: Is the `total` state the same as the `totalAmount` array?

File: imports/ui/components/CosmosErrors.jsx [hunk 5]
```diff
@@ -84,7 +84,7 @@ export default class CosmosErrors extends Component {
                         }
                         
                         this.state = {
-                            error: (error.codespace && error.code && errors[error.codespace][error.code]) || errors.sdk[1],
+                            error: (error.codespace && error.code && errors[error.codespace] && errors[error.codespace][error.code] && errors[error.codespace][error.code]) || "Unknown error",
```
Comment (kwunyeung) [inline imports/ui/components/CosmosErrors.jsx]: Only one ` && errors[error.codespace][error.code]` is needed. I was trying to tell you where to add the code.

File: CHANGELOG.md (hunk 1 above)
Comment (MonikaCat) [reply CHANGELOG.md]: Fixed in `b438c15` commit

File: imports/ui/accounts/Account.jsx (hunk 2 above)
Comment (MonikaCat) [reply imports/ui/accounts/Account.jsx]: I have applied suggested changes in `a9da9c0 ` commit

**Pull request merged**